### PR TITLE
[Concept Entry] SwiftUI Layouts

### LINKED ISSUE #2102: [Concept Entry] SwiftUI Layouts
### Reason/inspiration (optional)

We would like a new concept entry for SwiftUI Layouts.

The edit should go in an existing file in docs/content/swiftui/concepts/layouts/layouts.md 

The entry should:

- Explain what view layouts are in SwiftUI, starting with an introductory pararaph.
- Overview on custom layouts.
- Feel free to briefly mention available views in [view terms](https://www.codecademy.com/resources/docs/swiftui/views).

Please refer to the [entry template](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by gracekishino**
Add new concept entry for SwiftUI layouts.

### Description

Added new folder `layouts` inside `content/swiftui/concepts/`.
Added new entry file `content/swiftui/concepts/layouts/layouts.md` for SwiftUI Layouts.

Closes #2102

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swiftui/concepts/layouts/layouts.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  - 'iOS'
+Tags:
+  - 'SwiftUI'
+  - 'UI'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**`Layout`** is a `protocol` in [SwiftUI](https://www.codecademy.com/resources/docs/swiftui) used to create layouts for user interfaces. Layouts specify how to arrange the [views](https://www.codecademy.com/resources/docs/swiftui/views) they contain. It's possible to create custom and conditional layouts. There are also built-in layouts that can be used when creating conditional layouts.
```
Comment (SSwiniarski) [inline content/swiftui/concepts/layouts/layouts.md:15]:
```suggestion
**`Layout`** is a `protocol` in SwiftUI used to create layouts for user interfaces. Layouts specify how to arrange the [views](https://www.codecademy.com/resources/docs/swiftui/views) they contain. It's possible to create custom and conditional layouts. There are also built-in layouts that can be used when creating conditional layouts.
```
We generally don't link to parent entries. Docs navigation provides those automatically.

File: media/swiftui-conditional-layout.gif
Comment (SSwiniarski) [inline media/swiftui-conditional-layout.gif:1]: If this is supposed to animate, I am not seeing the animation.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@gracekishino I did a first review and only had a couple of comments.

File: media/swiftui-conditional-layout.gif
Comment (gracekishino) [reply media/swiftui-conditional-layout.gif:1]: @SSwiniarski the gif wasn't looping so now replaced it with a looping one. Hope you can see it OK now!

**Review by SSwiniarski — APPROVED** [review]
@gracekishino Looks good for a second review.

**Review by yangc95 — APPROVED** [review]
@gracekishino wow! This is excellent

**Comment by yangc95** [comment]
🥳 Outstanding @gracekishino live link: https://www.codecademy.com/resources/docs/swiftui/layouts
Screenshot 2023-08-03 at 4 50 37 PM

cc: @SSwiniarski

**Comment by gracekishino** [comment]
@yangc95 @SSwiniarski Thanks so much! This one did take me quite a while to research and understand so learned loads from it :)

**Pull request merged**